tests/integration/test_bundle.py

```diff
@@ -196,3 +199,33 @@ async def test_bundle_charms_can_handle_frequent_update_status(ops_test: OpsTest
 
     # After update-status frequency is restored to default, make sure all charms settle into active
     await ops_test.model.wait_for_idle(status="active")
+
+
+async def test_prometheus_scrapes_loki_through_traefik(ops_test: OpsTest):
+    """Prometheus should correctly scrape Loki through its traefik endpoint."""
+    prom_url = await get_proxied_unit_url(ops_test, "prometheus", 0)
+
+    response = urllib.request.urlopen(f"{prom_url}/api/v1/targets", data=None, timeout=2.0)
+    assert response.code == 200
+    targets = json.loads(response.read())["data"]["activeTargets"]
+    targets_summary = [(t["discoveredLabels"]["__metrics_path__"], t["health"]) for t in targets]
+    assert (f"/{ops_test.model.name}-loki-0/metrics", "up") in targets_summary
+    logger.info("prometheus is successfully scraping loki through traefik")
+
+
+async def test_loki_receives_logs_through_traefik(ops_test: OpsTest):
+    """Loki should be able to receive logs through its traefik endpoint."""
+    loki_url = await get_proxied_unit_url(ops_test, "loki", 0)
+
+    await ops_test.model.deploy("zinc-k8s", application_name="zinc", channel="stable", trust=True)
+    await ops_test.model.wait_for_idle(status="active")
+    # Create the relation
+    await ops_test.model.add_relation("loki", "zinc")
+    await ops_test.model.wait_for_idle(status="active")
+    # Check that logs are coming in
+    response = urllib.request.urlopen(f"{loki_url}/loki/api/v1/series")
+    assert response.code == 200
+    series = json.loads(response.read())["data"]
+    series_charms = [s["juju_charm"] for s in series]
+    assert "zinc-k8s" in series_charms
+    logger.info("loki is successfully receiving zinc logs through traefik")
```

Comment:
Is this `info` a leftover from debugging?

Reply:
Nope, it mimics what's done in all the other tests!